unicon_backend/lib/amqp.py

```diff
@@ -56,20 +69,38 @@ def stop(self):
         self._closing = True
         if self._chan:
             self._chan.close()
-        if self._conn:
+        if self._conn and not self._conn.is_closed:
             self._conn.close()
+        if self._reconnect_timer:
+            self._reconnect_timer.cancel()
+
+    def _reconnect(self):
```

Comment:
the reconnect logic seems a little unnecessarily complicated, especially as it introduce new threads in a async event loop

have you looked at: https://github.com/pika/pika/blob/802a625299707ba800d3da910afd506beb5a9462/examples/asynchronous_consumer_example.py#L393

Reply:
I will have a look at it. I think it's a good idea to separate out the logic to another class.